sql/core/src/main/scala/org/apache/spark/sql/execution/r/ArrowRRunner.scala

```diff
@@ -202,14 +175,17 @@ class ArrowRRunner(
               // Likewise, there looks no way to send each batch in streaming format via socket
               // connection. See ARROW-4512.
               // So, it reads the whole Arrow streaming-formatted binary at once for now.
-              val in = new ByteArrayReadableSeekableByteChannel(readByteArrayData(length))
+              val buffer = new Array[Byte](length)
```

Comment:
why replace `readByteArrayData`?

Reply:
Just realised that `readByteArrayData` is only two lines, and thought it's better to put all reading related logics from RRunner into `RRunner`'s `ReaderIterator` in `newReaderIterator`.